lib/signaling/participant.js

```diff
@@ -163,13 +163,14 @@ class ParticipantSignaling extends StateMachine {
    * updates the subscriber priority for the given track.
    * @param {Track.SID} trackSid
    * @param {?Track.Priority} priority
+   * @param {'publish'|'subscribe'} publishOrSubscribe
    * @returns {void}
    */
-  updateTrackPriority(trackSid, priority) {
+  updateTrackPriority(trackSid, priority, publishOrSubscribe) {
     if (this._trackPrioritySignaling) {
-      this._trackPrioritySignaling.sendTrackPriorityUpdate(trackSid, 'subscribe', priority);
+      this._trackPrioritySignaling.sendTrackPriorityUpdate(trackSid, publishOrSubscribe, priority);
```

Comment:
@makarandp0 I think whether to `publish` or `subscribe` can be initialized in the constructors of LocalParticipantV2 and RemoteParticipantV2:
```js
class LocalParticipantV2 {
  constructor() {
    Object.defineProperties(this, {
      _trackAction: {
        value: 'publish'
      }
    });
  }
}

class RemoteParticipantV2 {
  constructor() {
    Object.defineProperties(this, {
      _trackAction: {
        value: 'subscribe'
      }
    });
  }
}

class ParticipantSignaling {
  updateTrackPriority(trackSid, priority) {
    ...
    this._trackPrioritySignaling.sendTrackPriorityUpdate(trackSid, this._trackAction, priority);
    ...
    this._enqueuedPriorityUpdates.set(trackSid, priority);
  }
}
```
This way you don't have to store `publish` or `subscribe` in each enqueued entry.